clouddriver-openstack/src/test/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/ResizeOpenstackAtomicOperationValidatorSpec.groovy

```diff
@@ -0,0 +1,135 @@
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
+import org.springframework.validation.Errors
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class ResizeOpenstackAtomicOperationValidatorSpec extends Specification {
+
+  Errors errors
+  AccountCredentialsProvider provider
+  DeployOpenstackAtomicOperationValidator validator
+  OpenstackNamedAccountCredentials credentials
+  OpenstackCredentials credz
+  OpenstackClientProvider clientProvider
+
+  String account = 'foo'
+  String application = 'app1'
+  String region = 'r1'
+  String stack = 'stack1'
+  String freeFormDetails = 'test'
+  boolean disableRollback = false
+  int timeoutMins = 5
+  String instanceType = 'm1.small'
+  String image = 'ubuntu-latest'
+  int maxSize = 5
+  int minSize = 3
+  String networkId = '1234'
+  String poolId = '5678'
+  List<String> securityGroups = ['sg1']
+
+  def setup() {
+    clientProvider = Mock(OpenstackClientProvider)
+    clientProvider.getProperty('allRegions') >> ['r1']
+    GroovyMock(OpenstackProviderFactory, global: true)
+    OpenstackProviderFactory.createProvider(credentials) >> clientProvider
+    credz = new OpenstackCredentials(credentials)
+    errors = Mock(Errors)
+    credentials = Mock(OpenstackNamedAccountCredentials) {
+      _ * getCredentials() >> credz
+    }
+    provider = Mock(AccountCredentialsProvider) {
+      _ * getCredentials(_) >> credentials
+    }
+    validator = new DeployOpenstackAtomicOperationValidator(accountCredentialsProvider: provider)
+  }
+
+  def "Validate no error"() {
+    given:
+    ServerGroupParameters params = new ServerGroupParameters(instanceType: instanceType, image:image, maxSize: maxSize, minSize: minSize, networkId: networkId, poolId: poolId, securityGroups: securityGroups)
+    DeployOpenstackAtomicOperationDescription description = new DeployOpenstackAtomicOperationDescription(account: account, application: application, region: region, stack: stack, freeFormDetails: freeFormDetails, disableRollback: disableRollback, timeoutMins: timeoutMins, serverGroupParameters: params, credentials: credz)
+
+    when:
+    validator.validate([], description, errors)
+
+    then:
+    0 * errors.rejectValue(_,_)
+  }
+
+
+  @Unroll
+  def "Validate create missing required core field - #attribute"() {
+    given:
+    ServerGroupParameters params = new ServerGroupParameters(instanceType: instanceType, image:image, maxSize: maxSize, minSize: minSize, networkId: networkId, poolId: poolId, securityGroups: securityGroups)
+    DeployOpenstackAtomicOperationDescription description = new DeployOpenstackAtomicOperationDescription(account: account, application: application, region: region, stack: stack, freeFormDetails: freeFormDetails, disableRollback: disableRollback, timeoutMins: timeoutMins, serverGroupParameters: params, credentials: credz)
+    if (attribute != 'stack') {
+      description."$attribute" = ''
```

Comment:
`''` and `1-2-3` feels like it should be a 3rd item in the `where` clause. and actually make it a table now instead of using `var << []`